### PR TITLE
add solutions for ski course design (376)

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-376.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+<LanguageSection>
+
+<PySection>
+
+```python
+with open('skidesign.in') as read:
+	n = int(read.readline())
+	elevation = [int(read.readline()) for _ in range(n)]
+
+ans = 10 ** 10
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:23]: use inf

File: solutions/usaco-376.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+for i in range(1, 83):
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:25]: 83? seems like an oddly specific number

File: solutions/usaco-376.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	temp = 0
+	for j in elevation:
+		if j < i:
+			temp += (j - i) ** 2
+		elif j > i + 17:
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:30]: 17 as well

File: solutions/usaco-376.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+</PySection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int elevation[1001];
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:45]: (optional) just use a vector & dynamically size it

File: solutions/usaco-376.mdx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+int ans = 1e9;
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:46]:
```suggestion
int ans = INT32_MAX;
```
wait why is this global

File: solutions/usaco-376.mdx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	freopen("skidesign.out", "w", stdout);
+
+	int n;
+	cin >> n;
+
+	for (int i = 0; i < n; i++) {
+		cin >> elevation[i];
+	}
+	
+	for(int i = 0; i < 83; i++){
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:60]: spaces for control flow statements

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-376.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-376.mdx:25]: 100-17

File: solutions/usaco-376.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-376.mdx:30]: context of problem

File: solutions/usaco-376.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-376.mdx:25]: put 17 in a constant MAX_DIFF

Comment (jjliewie) [reply solutions/usaco-376.mdx:25]: okay

File: solutions/usaco-376.mdx (hunk 1 above)
Comment (jjliewie) [reply solutions/usaco-376.mdx:23]: alright

File: solutions/usaco-376.mdx [hunk 7]
```diff
@@ -0,0 +1,85 @@
+---
+id: usaco-376
+source: USACO Bronze 2014 January
+title: Ski Course Design
+author: Juheon Rhee
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_skidesign.html)
```
Comment (ryanchou-dev) [inline solutions/usaco-376.mdx:8]:
```suggestion
[Official Analysis (C++)](http://www.usaco.org/current/data/sol_skidesign.html)
```

**Review by ryanchou-dev — APPROVED** [review]
⭐

File: solutions/usaco-376.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-376.mdx:51]: bump

File: solutions/usaco-376.mdx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+```python
+MAX_HEIGHT = 100
+MAX_DIFF = 17
+	
+with open('skidesign.in') as read:
+	n = int(read.readline())
+	elevation = [int(read.readline()) for _ in range(n)]
+
+ans = float('inf')
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:26]: rename to `min_cost` (same in c++)

File: solutions/usaco-376.mdx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+for i in range(1, MAX_HEIGHT - MAX_DIFF):
+	temp = 0
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:29]: rename to `this_cost` (same in c++)

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-376.mdx (hunk 8 above)
Comment (jjliewie) [reply solutions/usaco-376.mdx]: done

File: solutions/usaco-376.mdx (hunk 9 above)
Comment (jjliewie) [reply solutions/usaco-376.mdx:29]: done

Comment (SansPapyrus683) [reply solutions/usaco-376.mdx:29]: really? i don't see anything

Comment (jjliewie) [reply solutions/usaco-376.mdx:29]: i changed ans to min_cost instead of this_cost

Comment (SansPapyrus683) [reply solutions/usaco-376.mdx:29]: wb `this_cost`

Comment (jjliewie) [reply solutions/usaco-376.mdx:29]: okay

File: solutions/usaco-376.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-376.mdx:51]: bump

File: solutions/usaco-376.mdx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+	int this_cost = INT32_MAX;
+	for(int i = 0; i < MAX_HEIGHT - MAX_DIFF; i++){
+		int temp = 0;
+		for (int j = 0; j < n; j++) {
+			if (elevation[j] < i) {
+				temp += (i - elevation[j]) * (i - elevation[j]);
+			}
+
+			if (elevation[j] > i + MAX_DIFF) {
+				temp += (elevation[j] - i - MAX_DIFF) * (elevation[j] - i - MAX_DIFF);
```
Comment (SansPapyrus683) [inline solutions/usaco-376.mdx:72]: over 80 chars?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-376.mdx (hunk 4 above)
Comment (jjliewie) [reply solutions/usaco-376.mdx:51]: no. i don't like vectors.

Comment (SansPapyrus683) [reply solutions/usaco-376.mdx:51]: elaborate?

**Pull request closed without merging**